### PR TITLE
fix confusing slit resolution parameters and add guardrails

### PR DESCRIPTION
**Pull request by pbeaucage**
As discussed at length in #1753, this adds several guardrails to prevent incorrect use of the 1D slit-smearing function.  Specifically, adds LOUD warnings if a user specifies a slit length that is less than the width, a slit geometry with an aspect ratio less than 10:1, or a slit length less than q_max for the data.

This also changes the nomenclature throughout from the confusing qx_width and qy_width to the canSAS standard q_length and q_width, where length is the long axis of the slit and width the short axis.

This depends on and is linked to issue sasview/sasmodels#528 and PR sasview/sasmodels#530

### CODE REVIEW

**Review by lucas-wilkins — APPROVED** [review]
Seems fine

**Review by butlerpd — CHANGES_REQUESTED** [review]
The severity of the log message (critical) is not really conveyed by the look of the popup. It there not a standard "dire" popup in QT? at least something with a red box with an x in it?  Also I'm wondering if the request in the issue that the length be larger than the qmax is not a bit aggressive?  Finally, I think that length is actually the half length of the slit? Even so, USANS would not actually qualify (and doubt USAXS would either?

I suggest we remove the requirement for length to be larger than qmax and merge this as better than before and then look more deeply into whether we should be limiting the qmax to smaller than length?  Also some more indication of the importance of the pop up would be good.

File: src/sas/qtgui/Perspectives/Fitting/SmearingWidget.py [hunk 1]
```diff
@@ -384,16 +386,48 @@ def onSlitSmear(self):
...
+        elif d_length < 10 * d_width:
+            msg = f'Slit length specified which is less than 10 x slit width. ' \
+                  f'This is not slit-smearing (at least not in the form we implement). \n' \
+                  f'Use pinhole smearing instead.'
+            smearing_error_flag = True
+        elif d_length < q_max:
+            msg = f'Length specified which is less than q_max for the data. \n' \
+                  f'This is not covered by existing smearing model. \n' \
+                  f'Use pinhole smearing instead. '
+            smearing_error_flag = True
```
Comment (pbeaucage) [inline src/sas/qtgui/Perspectives/Fitting/SmearingWidget.py:416]:
```suggestion
        # elif d_length < q_max:
        #     msg = f'Length specified which is less than q_max for the data. \n' \
        #           f'This is not covered by existing smearing model. \n' \
        #           f'Use pinhole smearing instead. '
        #    smearing_error_flag = True
```

**Review by butlerpd — APPROVED** [review]
works from development environment and cannot really test installer since it depends on the sasmodels as well so merging now.  We may want to look again at the popup later.

**Pull request merged**